src/main/java/net/snowflake/ingest/streaming/internal/OpenChannelRequestInternal.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+package net.snowflake.ingest.streaming.internal;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import net.snowflake.ingest.streaming.OpenChannelRequest;
+import net.snowflake.ingest.utils.Constants;
+import net.snowflake.ingest.utils.Utils;
+
+/** Class used to serialize the {@link OpenChannelRequest} */
+public class OpenChannelRequestInternal implements StreamingIngestRequest {
```

Comment:
same comment as DropChannelRequestInternal - why do we need an OpenChannelRequest and OpenChannelRequestInternal as two separate classes?

Reply:
nvm, i see how this is not meant to be a customer-exposed class thus we need this internal class.